python/cugraph/cugraph/gnn/graph_store.py

```diff
@@ -253,40 +254,35 @@ def sample_neighbors(
                 columns={"sources": src_n, "destinations": dst_n}, inplace=True
             )
 
-        # FIXME: Remove once below lands
-        # https://github.com/rapidsai/cugraph/issues/2444
-        edge_df = self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]]
-        sampled_df = edge_df.merge(sampled_df)
-
         return (
             sampled_df[src_n].to_dlpack(),
             sampled_df[dst_n].to_dlpack(),
-            sampled_df[eid_n].to_dlpack(),
+            sampled_df['indices'].to_dlpack(),
         )
 
     @cached_property
     def extracted_reverse_subgraph_without_renumbering(self):
         # TODO: Switch to extract_subgraph based on response on
         # https://github.com/rapidsai/cugraph/issues/2458
-        subset_df = self.gdata._edge_prop_dataframe[[src_n, dst_n]]
+        subset_df = self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]]
```

Comment:
Now that https://github.com/rapidsai/cugraph/pull/2444 is merged, are we able to start migrating code like this to calls to `get_edge_data()` ?

Reply:
Nope , we need a reverse graph in this case.  See  below line: 

https://github.com/rapidsai/cugraph/blob/cb3ca16ddd61f072c6addaff5d45e97dccff7be0/python/cugraph/cugraph/gnn/graph_store.py#L268


To remove this direct accessor we need to resolve either of these two issues.   

1. https://github.com/rapidsai/cugraph/issues/2458

OR

2. https://github.com/rapidsai/cugraph/issues/2468


For more context we think 2 might be easier to do based on internal discussion [here](https://nvidia.slack.com/archives/CV0B740TW/p1659033606208099) but would love your input towards it.

Reply:
Gotcha, thanks for the response.  I'll look into those conversations/issues you started.  I agree that option 2 would be better for your use case (mainly because it eliminates the need for an extra step and seems more self-documenting), but option 1 could be useful in general for other graph users so we may end up supporting both APIs.